File: imate_android_app/app/src/main/java/com/inty/imate/voicecall/VoiceCallModels.kt
package com.inty.imate.voicecall

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
enum class VoiceCallPacketType {
    @SerialName("audio") AUDIO,
    @SerialName("audio_response") AUDIO_RESPONSE,
    @SerialName("status") STATUS,
    @SerialName("error") ERROR,
    @SerialName("session_info") SESSION_INFO,
    @SerialName("transcript") TRANSCRIPT,
    @SerialName("user_transcript") USER_TRANSCRIPT,
    @SerialName("end") END,
    UNKNOWN,
}

@Serializable
data class VoiceCallPacket(
    val type: VoiceCallPacketType = VoiceCallPacketType.UNKNOWN,
    val data: String? = null,
    val status: String? = null,
    val message: String? = null,
    @SerialName("error_code") val errorCode: String? = null,
    @SerialName("remaining_duration") val remainingDuration: Int? = null,
)

enum class VoiceCallConnectionState {
    DISCONNECTED,
    CONNECTING,
    CONNECTED,
    ERROR,
}

File: imate_ios_app/imate/voicecall/VoiceCallModels.swift
import Foundation

enum VoiceCallPacketType: String, Codable {
    case audio = "audio"
    case audioResponse = "audio_response"
    case status
    case error
    case sessionInfo = "session_info"
    case transcript
    case userTranscript = "user_transcript"
    case end
}

struct VoiceCallPacket: Codable {
    var type: VoiceCallPacketType
    var data: String?
    var status: String?
    var message: String?
    var errorCode: String?
    var remainingDuration: Int?

    enum CodingKeys: String, CodingKey {
        case type
        case data
        case status
        case message
        case errorCode = "error_code"
        case remainingDuration = "remaining_duration"
    }
}

enum VoiceCallConnectionState: String {
    case disconnected
    case connecting
    case connected
    case error
}
